packages/create-sitecore-jss/src/templates/angular-sxp/src/app/components/styleguide-custom-route-type/styleguide-custom-route-type.component.ts

```diff
@@ -1,22 +1,23 @@
-import { Component, OnInit, OnDestroy, Input } from '@angular/core';
+import { Component, OnInit, OnDestroy, Input, inject } from '@angular/core';
 import { Subscription } from 'rxjs';
 import { JssContextService } from '../../jss-context.service';
-import { ComponentRendering } from '@sitecore-jss/sitecore-jss-angular';
+import { ComponentRendering, JssModule } from '@sitecore-jss/sitecore-jss-angular';
+import { StyleguideSpecimenComponent } from '../shared/styleguide-specimen/styleguide-specimen.component';
 
 @Component({
   selector: 'app-styleguide-custom-route-type',
   templateUrl: './styleguide-custom-route-type.component.html',
+  imports: [JssModule, StyleguideSpecimenComponent]
 })
 export class StyleguideCustomRouteTypeComponent implements OnInit, OnDestroy {
   @Input() rendering: ComponentRendering;
   contextFields: { [name: string]: unknown };
 
   private contextSubscription: Subscription;
-
-  constructor(private jssContext: JssContextService) { }
+  private jssContext = inject(JssContextService);
 
   ngOnInit() {
-    this.contextSubscription = this.jssContext.state.subscribe((state) => {
+    this.contextSubscription = this.jssContext.state.subscribe((state: { sitecore: { route: { fields: { [name: string]: unknown } } } }) => {
```

Comment:
Can you import `LayoutServiceData` from `sitecore-jss`? so it will not be required to hardcode the type